src/shared/ReentrancyGuard.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
```

Comment:
Why are we adding this to our code? why can't we just import it?

Reply:
It's refactored to be used with our AppStorage storage pattern. It's not recommended we have any storage variables local to a contract since we are delegatecalling into them and therefore confusing which storage slot we are writing into.